labelling-wc-apps/README.md

```diff
@@ -0,0 +1,54 @@
+# Labelling workload cluster's apps
+
+The purpose of this RFC is to propose a labelling scheme to safely identify a workload cluster's defining apps.
+
+This is required for:
+
+- deleting a workload cluster on CAPI;
+- displaying a workload cluster's defining apps in Happa.
+
+## Deleting a workload cluster on CAPI
+
+To delete a workload cluster on CAPI we need to identify corresponding cluster app.
+
+### Current state
+
+On CAPI we are relying on the resource names (in the org namespace) for cluster app selection which is not recommended. There is no reference between cluster CR and cluster app CR - unlike other cluster's app resources, cluster app CR doesn't have `giantswarm.io/cluster` label. It cannot be set at the moment, since it will break cluster deletion.
+
+### Proposal
+
+Require `giantswarm.io/cluster` to be present in all app resources related to a workload cluster, including cluster app.
+
+Introduce a new `giantswarm.io/cluster-app` label to safly mark cluster app CRs.
+
+Refactor places where it's assumed that cluster app doesn't have `giantswarm.io/cluster` label to rely on a new `giantswarm.io/cluster-app` label instead.
+
+## Displaying a workload cluster's defining apps in Happa
+
+In Happa, we display workload cluster's apps in two separate lists - apps installed by a user and default (preinstalled) apps. To achieve that, we need:
+
+- fetch all App CRs related to a workload cluster. On CAPI this list should not contain cluster app App CR.
```

Comment:
Why should this list not include the cluster app?

Reply:
I should rephrase it. It's more like 
> Currently we don't display cluster app in the UI, so we should either fetch all the apps except cluster app, or we need a way to filter it out later.

Reply:
Now I get it. Yep, would be good to clarify that.